src/Integrations/POSCatalog/ProductMapper.php

```diff
@@ -52,7 +77,12 @@ public function map_product( WC_Product $product ): array {
 		 * @param array      $row     Mapped product data.
 		 * @param WC_Product $product Product object.
 		 */
-		return apply_filters( 'oapfw_map_catalog_product', $row, $product );
+		$row = apply_filters( 'oapfw_map_catalog_product', $row, $product );
+
+		if ( null === $this->fields ) {
+			return $row;
+		}
+		return _rest_array_intersect_key_recursive( $row, $this->fields );
```

Comment:
Do we need recursion here? From the issue description, it seems that `array_intersect_key` is quite enough. 

```suggestion
		return array_intersect_key( $row, $this->fields );
```